### PR TITLE
Add copy commands

### PR DESCRIPTION
**Pull request by jonasongg**
addresses #68 and #70

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Person.java [hunk 1]
```diff
@@ -33,4 +33,6 @@ public Name getName() {
 
     public abstract String toString();
 
+    public abstract String detailsToCopy();
```
Comment (matochichap) [inline src/main/java/seedu/address/model/person/Person.java:36]: maybe add javadoc for this method?

File: src/test/java/seedu/address/logic/commands/CopyApplicantCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Applicant;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code CopyApplicantCommand}.
+ */
+public class CopyApplicantCommandTest {
```
Comment (matochichap) [inline src/test/java/seedu/address/logic/commands/CopyApplicantCommandTest.java:33]: nice test cases!

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 3]
```diff
@@ -82,17 +86,36 @@ public void parseCommand_help() throws Exception {
         assertTrue(parser.parseCommand(HelpCommand.COMMAND_WORD + " 3") instanceof HelpCommand);
     }
 
-    // TODO: adapt for ViewMemberCommand and ViewApplicantCommand
-    //    @Test
-    //    public void parseCommand_list() throws Exception {
-    //        assertTrue(parser.parseCommand(ListCommand.COMMAND_WORD) instanceof ListCommand);
-    //        assertTrue(parser.parseCommand(ListCommand.COMMAND_WORD + " 3") instanceof ListCommand);
-    //    }
+    @Test
```
Comment (matochichap) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:107]: thanks for adding the tests for this class

File: src/main/java/seedu/address/logic/commands/CopyMemberCommand.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Member;
+
+/**
+ * Copies a member from the address book to the clipboard.
+ */
+public class CopyMemberCommand extends Command {
+
+    public static final String COMMAND_WORD = "copyMember";
```
Comment (matochichap) [inline src/main/java/seedu/address/logic/commands/CopyMemberCommand.java:22]: should we not restrict users to type copyMember and allow them to type a command with any number of upper and lower case? e.g. CopyMember is also allowed

**Review by matochichap — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/address/logic/commands/CopyApplicantCommand.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+/**
+ * Copies an applicant from the address book to the clipboard.
+ */
+public class CopyApplicantCommand extends Command {
+
+    public static final String COMMAND_WORD = "copyApplicant";
+    public static final String COMMAND_ALIAS = "cpa";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Copies the details of the applicant identified by the index number used"
```
Comment (Saezenn) [inline src/main/java/seedu/address/logic/commands/CopyApplicantCommand.java]: Phrasing is a bit weird. Perhaps can change to : Copies the name and phone number of the applicant at the specified index to clipboard

File: src/main/java/seedu/address/logic/commands/CopyMemberCommand.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    public static final String COMMAND_ALIAS = "cpm";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
```
Comment (Saezenn) [inline src/main/java/seedu/address/logic/commands/CopyMemberCommand.java:25]: same for this as mentioned earlier in cpa. 

Perhaps can change to : Copies the details (name, phone number, email, etc) of the applicant at the specified index to clipboard.

**Review by Saezenn — COMMENTED** [review]
Overall, LGTM. Can consider making a copyAll feature to copy all details (entire list)

File: src/main/java/seedu/address/logic/commands/CopyApplicantCommand.java (hunk 5 above)
Comment (jonasongg) [reply src/main/java/seedu/address/logic/commands/CopyApplicantCommand.java]: good catch! ive changed it

File: src/main/java/seedu/address/logic/commands/CopyMemberCommand.java (hunk 6 above)
Comment (jonasongg) [reply src/main/java/seedu/address/logic/commands/CopyMemberCommand.java:25]: same as above

**Comment by jonasongg** [comment]
@Saezenn thanks for the comments! i've resolved it in the latest commit. might want to merge this first and leave the mass copy for another PR because my edits to the DG that includes the copy commands are already merged!

**Pull request merged**